### PR TITLE
Allow using enums with empty string value with EnumType

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by LordSimal** [comment]
With that if one makes an enum associated column nullable then both `null` and `''` are valid values. I don't like that.

**Comment by ADmad** [comment]
Having both empty string and null as valid values for an enum column sounds like an edge case.

My change allows one to either have an empty string as a valid enum value **or** allow the column to be nullable.

My use case is that I have a column where an empty string is used to indicate "no value" and I want to display "No selection" in the frontend. Using `null` instead of an empty string would make things unnecessarily cumbersome. Instead of just doing `$value->label()` I would have to add null check and hard code the `'No selection'` string for display. In forms too instead of having all options auto generated I would have to manually add `No selection` to the options list.

**Comment by LordSimal** [comment]
I can see the benefit related to forms. Even though this would introduce a behavior change I think this is fine for 5.1

**Review by LordSimal — APPROVED** [review]

File: src/Database/Type/EnumType.php [hunk 1]
```diff
@@ -167,7 +167,7 @@ public function toStatement(mixed $value, Driver $driver): int
      */
     public function marshal(mixed $value): ?BackedEnum
     {
-        if ($value === null || $value === '') {
+        if ($value === null) {
```
Comment (dereuromark) [inline src/Database/Type/EnumType.php:170]: I am fairly certain we added some of this for compatibility with int backed enums and empty values 
Will have to check that there arent regressions in that direction

Comment (ADmad) [reply src/Database/Type/EnumType.php:170]: Empty string to int backed enum is still accounted for (check line 187-190) and returns `null` as before. There are also tests for the same.

**Pull request merged**